### DISCUSSION TITLE
Help calling set_tou_periods in a NodeRed flow

### DISCUSSION THREAD

**Discussion by billerby** [wlcrs/huawei_solar#215] [General]
Hi!

I am trying to use the new set_tou_service from the release candidate but need a little help figuring out how to configure my service node. I have configured the data string to send to my luna battery in a function node before, it a string in the msg.payload: `"'22:00-03:00/1234567/+'"`

The call service node have the following options:
![image](https://user-images.githubusercontent.com/123237/209541270-748851b7-b5dd-47f4-924e-763355e4ba31.png)

Should I provide the device_id in the Data? What do I need to enter in the data field? Anyone have a clue?

This is the rest of the flow btw:
![image](https://user-images.githubusercontent.com/123237/209541448-1671e215-9d77-4f6d-9108-926625d37d81.png)

Appreciate any help :)

**Comment by wlcrs** [comment]
Start with using the 'Services' tab in the Home Assistant developers tools (wrench icon above the link to the settings) to play around with the service. Use the GUI to fill in the correct values and switch to YAML mode to see what you need to input in the payload in Node Red

**Reply by billerby** [reply]
Great tip! Thanks, noticed something that I maybe should report as a bug?

If I send the following "22:00:-03:00" its not accepted because 
`huawei_solar.exceptions.TimeOfUsePeriodsException: TOU period is invalid (start-time is greater than end-time)`

One can have differnt opinions on if thats true :)

**Reply by wlcrs** [reply]
You'll need to split that into two periods.
